src/Engine/ProtoCore/DSASM/Executive.cs

```diff
@@ -2308,15 +2319,35 @@ private void Execute(int exeblock, int entry, Language language = Language.NotSp
             }
         }
 
+        private int cacheBlockID = -1;
+        private int cacheClassIndex = -1;
+        private int cachesymbolIndex = -1;
+        private SymbolNode cachedSymbol = null;
+
         protected SymbolNode GetSymbolNode(int blockId, int classIndex, int symbolIndex)
         {
+            if (blockId == cacheBlockID && classIndex == cacheClassIndex && symbolIndex == cachesymbolIndex && cachedSymbol != null)
+            {
+                return cachedSymbol;
```

Comment:
How much is caching really going to help here? After all you're simply avoiding array/list indexing (`exe.runtimeSymbols[blockId].symbolList[symbolIndex]`), which I believe should just be `O(1)` operations, or am I missing something?